### PR TITLE
use only locations supported by the required resource providers on azure

### PR DESCRIPTION
**Pull request by lpuskas**
* To avoid restricted regions on azure (that result in erroneous pages on the UI) locations that are not supported by the `Microsoft.Compute` resource provider and `"locations/vmSizes"` resource types are excluded.
* refreshed some azure sdk libs, added a new package

### CODE REVIEW

File: pkg/productinfo/azure/productinfo_azure.go [hunk 1]
```diff
@@ -345,15 +351,46 @@ func (a *AzureInfoer) GetZones(region string) ([]string, error) {
...
+		resourceType      = "locations/vmSizes"
+	)
+
+	if providers, err := a.providersClient.Get(context.TODO(), providerNamespace, ""); err == nil {
+		for _, pr := range *providers.ResourceTypes {
+			if *pr.ResourceType == resourceType {
+				for _, displName := range *pr.Locations {
+					if loc, ok := allLocations[displName]; ok {
+						log.Debugf("found supported location. [name, display name] = [%s, %s]", loc, displName)
+						supLocations[loc] = loc
```
Comment (martonsereg) [inline pkg/productinfo/azure/productinfo_azure.go]: this should be `supLocations[loc] = displName` to store the display name as well

File: pkg/productinfo/azure/productinfo_azure.go [hunk 2]
```diff
@@ -345,15 +351,46 @@ func (a *AzureInfoer) GetZones(region string) ([]string, error) {
...
+					} else {
+						log.Debugf("unsupported location. [name, display name] = [%s, %s]", loc, displName)
+					}
+				}
```
Comment (martonsereg) [inline pkg/productinfo/azure/productinfo_azure.go]: we can break the loop here

File: pkg/productinfo/azure/productinfo_azure.go (hunk 1 above)
Comment (lpuskas) [reply pkg/productinfo/azure/productinfo_azure.go]: thanks, that was the intention :)

File: pkg/productinfo/azure/productinfo_azure.go (hunk 2 above)
Comment (lpuskas) [reply pkg/productinfo/azure/productinfo_azure.go]: 👍

**Pull request merged**